### PR TITLE
Fixes #6: Replace find-java-home with @viperproject/locate-java-home

### LINKED ISSUE #6: `findJavaHome` returns wrong directory on macOS
I run macOS 12.6.5 and have an old `openjdk 11.0.10 2021-01-19` java version installed. `findJavaHome()` seems to return `/usr` as my `JAVA_HOME` instead of `/Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home`, which is my actual `JAVA_HOME`. This breaks `sdkmanager` so that it doesn‘t install packages, but it also doesn‘t exit.

Apparently, on macOS it is pretty easy to find the `JAVA_HOME` using `/usr/libexec/java_home` (https://stackoverflow.com/questions/22842743/how-to-set-java-home-environment-variable-on-mac-os-x-10-9)

### PR DESCRIPTION
**Pull request by zner0L**
Somehow the package of `locate-java-home` is broken and I already spent to much time trying to fix it. If you have any idea for a better fix, feel free.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,9 @@
 {
     "extends": "@baltpeter/tsconfig",
     "include": ["src/**/*"],
+    "compilerOptions": {
+        "esModuleInterop": true
+    },
```
Comment (baltpeter) [inline tsconfig.json]: That is redundant: https://github.com/tsconfig/bases/blob/014e8927b7155640f65a7492fb4bb09f5c49f531/bases/strictest.json#L17

**Comment by baltpeter** [comment]
This unfortunately breaks the CLI for me:

```
node:internal/process/esm_loader:97
    internalBinding('errors').triggerUncaughtException(
                              ^

Error [ERR_UNSUPPORTED_DIR_IMPORT]: Directory import '/home/benni/coding/JS/tweasel/andromatic/node_modules/@viperproject/locate-java-home/js/es6' is not supported resolving ES modules imported from /home/benni/coding/JS/tweasel/andromatic/dist/cli-install.js
Did you mean to import @viperproject/locate-java-home/js/es6/index.js?
    at new NodeError (node:internal/errors:399:5)
    at finalizeResolution (node:internal/modules/esm/resolve:319:17)
    at moduleResolve (node:internal/modules/esm/resolve:945:10)
    at defaultResolve (node:internal/modules/esm/resolve:1153:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:838:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:77:40)
    at link (node:internal/modules/esm/module_job:76:36) {
  code: 'ERR_UNSUPPORTED_DIR_IMPORT',
  url: 'file:///home/benni/coding/JS/tweasel/andromatic/node_modules/@viperproject/locate-java-home/js/es6'
}
```

**Comment by baltpeter** [comment]
Any reason you're not just doing `import locateJavaHomeModule from '@viperproject/locate-java-home';`? That uses the `es5` version but does work fine for me.

**Comment by baltpeter** [comment]
Just for reference, doing `import @viperproject/locate-java-home/js/es6/index.js` as the error suggests didn't work:

```
(node:2871701) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)
/home/benni/coding/JS/tweasel/andromatic/node_modules/@viperproject/locate-java-home/js/es6/index.js:1
import LocateJavaHome from './lib/locate_java_home';
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at internalCompileFunction (node:internal/vm:73:18)
    at wrapSafe (node:internal/modules/cjs/loader:1178:20)
    at Module._compile (node:internal/modules/cjs/loader:1220:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:169:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)
```

**Review by baltpeter — CHANGES_REQUESTED** [review]

**Comment by baltpeter** [comment]
Other than those two problems, this did resolve the issue on macOS for me. I was able to run this successfully (which previously would have gotten stuck):

```sh
npm i ./<`yarn pack` with the changes in this PR>.tgz
npm i appstraction
```

File: tsconfig.json (hunk 1 above)
Comment (zner0L) [reply tsconfig.json]: It was throwing errors for me before, but it seems like that was a caching issue.

**Review by baltpeter — APPROVED** [review]

**Pull request merged**